torchtune/modules/attention.py

```diff
@@ -306,3 +307,291 @@ def forward(
         # reshape the output to be the same shape as the input
         output = output.transpose(1, 2).contiguous().view(b, s_x, -1)
         return self.output_proj(output)
+
+
+class Gemma2Attention(nn.Module):
```

Comment:
I think we should consider putting this in `models/gemma2/_attention.py` instead since it's specific to the Gemma 2 model.